File: backend/src/main/java/com/sprouthub/sprouthub/service/TutorialService.java
package com.sprouthub.sprouthub.service;

import com.sprouthub.sprouthub.model.Tutorial;

import java.util.List;

public interface TutorialService {
    Tutorial createTutorial(Tutorial tutorial);
    Tutorial getTutorialById(String id);
    List<Tutorial> getAllTutorials();
    Tutorial updateTutorial(String id, Tutorial tutorial);
    void deleteTutorial(String id);
}

File: frontend/src/services/TutorialService.js
import api from '../utils/api';

const getAllTutorials = () => {
  return api.get('/tutorials');
};

const getTutorialById = (id) => {
  return api.get(`/tutorials/${id}`);
};

const createTutorial = (data) => {
  return api.post('/tutorials', data);
};

const updateTutorial = (id, data) => {
  return api.put(`/tutorials/${id}`, data);
};

const deleteTutorial = (id) => {
  return api.delete(`/tutorials/${id}`);
};

const tutorialService = {
  getAllTutorials,
  getTutorialById,
  createTutorial,
  updateTutorial,
  deleteTutorial,
};

export default tutorialService;